tests/integration/_common.py

```diff
@@ -206,23 +212,72 @@ def __enter__(self):
         # Figure out where the tox-initiated venv is. In environments where a
         # venv is activated as part of bashrc, $VIRTUAL_ENV won't be what we
         # expect inside of tmux, so we can't depend on it. We *must* determine
-        # it before we enter tmux.
+        # it before we enter tmux. Do this before we switch to bash
         venv_path = os.environ.get("VIRTUAL_ENV")
         if venv_path is None or ".tox" not in venv_path:
             raise AssertionError(
                 "VIRTUAL_ENV environment variable was not set but tox should have set it."
             )
         venv = os.path.join(shlex.quote(venv_path), "bin", "activate")
+        # get the USER before we start a clean shell
+        user = os.environ.get("USER")
+        home = os.environ.get("HOME")
+
+        # get a clean shell and predicatble prompt
+        self._cli_prompt = self._get_cli_prompt()
 
-        # send the config envvar + other set up commands
-        venv = f"source {venv}"
-        navigator_config = f"export ANSIBLE_NAVIGATOR_CONFIG={self._config_path}"
-        set_up_commands = [venv, navigator_config] + self._setup_commands
+        # set envars for this session
+        tmux_common = [f". {venv}"]
+        tmux_common.append(f"export HOME={home}")
+        tmux_common.append(f"export USER={user}")
+        tmux_common.append(f"export ANSIBLE_NAVIGATOR_CONFIG={self._config_path}")
```

Comment:
These paths should probably be quoted/escaped in case they ever have a space in them. But this is test code, so maybe not a huge deal.

Reply:
quoted